click/_compat.py

```diff
@@ -258,6 +258,18 @@ def filename_to_ui(value):
         if isinstance(value, bytes):
             value = value.decode(get_filesystem_encoding(), 'replace')
         return value
+
+    def which(cmd):
+        with open(os.devnull, 'w') as devnull:
+            p = subprocess.Popen(
+                ['which', cmd],
+                stdout=subprocess.PIPE,
```

Comment:
Can use `check_output` instead of `Popen.wait`.

Reply:
Done.